test/utils/utils.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import {
+	ByRoleMatcher,
+	ByRoleOptions,
+	GetAllBy,
+	queries,
+	queryHelpers,
+	screen as rtlScreen,
+	within as rtlWithin,
+	type Screen
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { filter } from 'lodash';
+
+export type UserEvent = ReturnType<(typeof userEvent)['setup']>;
+
+type ByRoleWithIconOptions = ByRoleOptions & {
+	icon: string | RegExp;
+};
+/**
+ * Matcher function to search an icon button through the icon data-testid
+ */
+const queryAllByRoleWithIcon: GetAllBy<[ByRoleMatcher, ByRoleWithIconOptions]> = (
+	container,
+	role,
+	{ icon, ...options }
+) =>
+	filter(
+		rtlWithin(container).queryAllByRole(role, options),
+		(element) => rtlWithin(element).queryByTestId(icon) !== null
+	);
+const getByRoleWithIconMultipleError = (
+	container: Element | null,
+	role: ByRoleMatcher,
+	options: ByRoleWithIconOptions
+): string => `Found multiple elements with role ${role} and icon ${options.icon}`;
+const getByRoleWithIconMissingError = (
+	container: Element | null,
+	role: ByRoleMatcher,
+	options: ByRoleWithIconOptions
+): string => `Unable to find an element with role ${role} and icon ${options.icon}`;
+
+const [getByRoleWithIcon] = queryHelpers.buildQueries<[ByRoleMatcher, ByRoleWithIconOptions]>(
```

Comment:
Is this necessary? Every icon should have already an accessible test id

Reply:
it is also possible to find the icon just with a findByTestId, but i find this implementation to be safer as it is making sure that the icon is clickable and that there are no multiple occurrences of it